src/components/TransactionItem/Details/Details.js

```diff
@@ -1,62 +1,146 @@
 // @flow
 
-import React from 'react'
+import React, { PureComponent } from 'react'
 import classNames from 'classnames'
 import { assoc } from 'ramda'
 
 import config from 'config'
 import handle from 'utils/eventHandlers/handle'
-import { JRaisedButton, JText } from 'components/base'
+import { JText, JIcon } from 'components/base'
+import TransactionItemMessage from './Message'
 
 type Props = {|
-  repeat: Function,
-  data: Transaction,
-  assetSymbol: ?string,
-  isActive: boolean,
+  +addFavorite: Function,
+  +repeat: Function,
+  +data: TransactionWithAssetAddress,
+  +assetDecimals: number,
+  +assetSymbol: string,
+  +isActive: boolean,
+|}
+
+type StateProps = {|
+  isMessage: boolean,
 |}
 
 const getTxLink = (txHash: Hash) => `${config.blockExplorerLink}/tx/${txHash}`
```

Comment:
need to create `utils/transactions/getTxLink` and getAddressLink
they should have 2 args - value: string, networkId: NetworkId

Reply:
need to get subdomain for api from here:
https://github.com/jibrelnetwork/jwallet-web/blob/master/src/services/etherscan.js#L9-L14
this map should be placed in src/config.js for re-usage